internal/overlay_repository.bzl

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _http_archive_impl(ctx):
+  if not ctx.attr.url and not ctx.attr.urls:
+    fail("either 'url' or 'urls' must be specified")
+  if ctx.attr.url and ctx.attr.urls:
+    fail("'url' and 'urls' may not both be specified")
+
+  urls = [ctx.attr.url] if ctx.attr.url else ctx.attr.urls
+  ctx.download_and_extract(
+      url = urls,
+      sha256 = ctx.attr.sha256,
+      type = ctx.attr.type,
+      stripPrefix = ctx.attr.strip_prefix,
+  )
+
+  _apply_overlay(ctx, ctx.attr.overlay)
+
+http_archive = repository_rule(
+    implementation = _http_archive_impl,
+    attrs = {
+        "url": attr.string(),
```

Comment:
I would drop url, it's been deprectated an awfully long time, no need to be backwards compatible.